tests/integration/golang/helpers/test.go

```diff
@@ -85,6 +95,80 @@ func (s *BaseTestSuite) SetupTest() {
 	s.Require().Nil(err)
 	s.TagFixtures = tagFixtures
 
-	// by default, unload everything.
-	s.Require().Nil(s.NamespaceFixtures.UnloadFixtures())
+	s.AddSetupHook(s.setup)
+	s.AddTearDownHook(s.tearDown)
+}
+
+func (s *BaseTestSuite) setup() {
+	s.Require().Nil(s.NamespaceFixtures.TruncateTables())
+
+	if !s.SkipCreateDefaultNamespace {
+		var err error
+		s.DefaultNamespace, err = s.NamespaceFixtures.CreateNamespace(context.Background(), &models.Namespace{
+			Code:                "default",
+			DefaultExperimentID: common.GetPointer(int32(0)),
+		})
+		s.Require().Nil(err)
+
+		if !s.SkipCreateDefaultExperiment {
+			s.DefaultExperiment, err = s.ExperimentFixtures.CreateExperiment(context.Background(), &models.Experiment{
+				Name:           "Default",
+				LifecycleStage: models.LifecycleStageActive,
+				NamespaceID:    s.DefaultNamespace.ID,
+			})
+			s.Require().Nil(err)
+
+			s.DefaultNamespace.DefaultExperimentID = s.DefaultExperiment.ID
+			_, err = s.NamespaceFixtures.UpdateNamespace(context.Background(), s.DefaultNamespace)
+			s.Require().Nil(err)
+		}
+	}
+}
+
+func (s *BaseTestSuite) AddSetupHook(hook func()) {
+	s.setupHooks = append(s.setupHooks, hook)
+}
+
+func (s *BaseTestSuite) runSetupHooks() {
+	for _, hook := range s.setupHooks {
+		hook()
+	}
+}
+
+func (s *BaseTestSuite) SetupTest() {
+	if !s.ResetOnSubTest {
+		s.runSetupHooks()
+	}
+}
```

Comment:
let's maybe group private and public function. now a bit hard to navigate over them. maybe let's have next grouping:
- firstly have private functions.
- then public functions.